### ISSUE TITLE
DBeaver: `NumberFormatException` when importing data into table with boolean column

### ISSUE THREAD

**Issue by hlcianfagna** [crate/crate-clients-tools#112]
```sql
create table repro (flag BOOLEAN);
```
Then create a `.csv` file and try to import it from DBeaver with the default "Data load settings" with "Skip bind values during insert" not selected
```
flag
0
```
In CrateDB this will fail with
```
NumberFormatException in java.base/java.math.BigDecimal.<init>(BigDecimal.java:703).CXX000.FSQLExceptions.java.ResToCrateException.L152.Wio.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:152)
```
in PostgreSQL it works
(DBeaver also succeeds if the column is `INT` instead of `BOOLEAN` or if we select the option to skip binding)

I can get the same stack trace in Java if I use a `setBigDecimal` against a `PreparedStatement` instead of a `setBoolean` , however this fails also against pgqsl while DBeaver works against pgsql.

So perhaps this something about `pg_type` not having composite type entries for the tables or about `pg_description` not having an entry for the `boolean` data type?

**Comment by amotl** [comment]
Hi @hlcianfagna and @surister. Can I humbly ask you to evaluate the problem closer, and maybe derive or move this issue into crate/crate, when applicable?

/cc @proddata, @seut

**Comment by hlcianfagna** [comment]
I did a quick test, this still occurs with the latest version of DBeaver and the latest nightly of CrateDB.

**Comment by amotl** [comment]
In the spirit of the above proposal, to focus on the CrateDB JDBC Driver again, which hopefully works in more occasions than the one for PostgreSQL, let me reference DBeaver plugin descriptions of [CrateDB (Legacy)](https://github.com/dbeaver/dbeaver/blob/5bb21a414dfe3a43d2401f9b313a666a1497146c/plugins/org.jkiss.dbeaver.ext.generic/plugin.xml#L1099-L1114) vs. [CrateDB](https://github.com/dbeaver/dbeaver/blob/5bb21a414dfe3a43d2401f9b313a666a1497146c/plugins/org.jkiss.dbeaver.ext.postgresql/plugin.xml#L609-L633).

Because always being compatible with vanilla PostgreSQL is difficult, maybe we should remove the PostgreSQL-based item again, and also promote "CrateDB (Legacy)" to canonical "CrateDB" here, to reduce confusion and friction for all users of DBeaver and CrateDB.

This might partly revert the updates submitted by @proddata the other day,

> Also this moves the generic CrateDB connector using the CrateDB specific JDBC Driver to legacy, as with recent CrateDB versions the default recommendations is to use the standard PostgreSQL JDBC Driver.
>
> - https://github.com/dbeaver/dbeaver/pull/20549

so I'd also like to involve him into this discussion, in order to possibly collectively agree to the conclusion that the CrateDB JDBC Driver should NOT be declared legacy, but should be promoted as the primary JDBC connector library to use with CrateDB.

/cc @kneth, @surister

**Comment by proddata** [comment]
> agree to the conclusion that the CrateDB JDBC Driver should NOT be declared legacy, but should be promoted and avertised to be the primary JDBC connector library to use with CrateDB instead.

> [!Caution]
> **The CrateDB JDBC driver is outdated and shall only be used with caution. Until there is a decision to bring this back to full maintenance, no tool or integration shall use this at all.**

If there is something not working in DBeaver, this shall be fixed by customizing the CrateDB connection there (which is possible) or adding functionality to CrateDB.

**Comment by proddata** [comment]
This issue has been fixed with [CrateDB 5.9.1](https://cratedb.com/docs/crate/reference/en/latest/appendices/release-notes/5.9.1.html) and later versions

> Fixed an issue that would cause an error to be thrown when using [PostgreSQL wire protocol](https://cratedb.com/docs/crate/reference/en/latest/interfaces/postgres.html#interface-postgresql) and attempting to insert 0 value for a numeric [NUMERIC](https://cratedb.com/docs/crate/reference/en/latest/general/ddl/data-types.html#type-numeric), or trying to get 0 numeric value from a query, e.g.:

```sql
cr> create table repro (flag BOOLEAN);
CREATE OK, 1 row affected (0.164 sec)
cr> SELECT * FROM repro;
+------+
| flag |
+------+
+------+
SELECT 0 rows in set (0.064 sec)

cr> SELECT * FROM repro;
+-------+
| flag  |
+-------+
| FALSE |
+-------+
SELECT 1 row in set (0.013 sec)
```

**Comment by amotl** [comment]
This is great news, thank you very much. In this case, all what is left is to wrap up this patch, demonstrating DBeaver with CrateDB >= 5.9.1, and using the vanilla pgJDBC adapter? Excellent!

- https://github.com/crate/cratedb-guide/pull/157
